dev-env/bgp/config.yaml.tmpl

```diff
@@ -16,4 +16,4 @@ data:
       protocol: bgp
       addresses:
       - 192.168.10.0/24
-      - fc00:f853:0ccd:e799::/64
+      - fc00:f853:0ccd:e799::/124
```

Comment:
but a `/64` is a valid configuration, right?  So is this just changing the test so that it doesn't hit a known bug?  :)

Reply:
It's a valid configuration that translates into maxint64 as value for the metric (see https://github.com/metallb/metallb/blob/997a177a762c6c8215ca9438083545529b675376/internal/allocator/allocator.go#L326), and already covered by unit tests: https://github.com/metallb/metallb/blob/4de2c0c7fbd3e4eef624bdff40c1c02f98d252d8/internal/allocator/allocator_test.go#L1213

I found more reasonable to have an exact number for the initial test.

When https://github.com/metallb/metallb/pull/931 lands, we can have multiple configurations testing that corner case, or maybe not since it would be more expensive in terms of e2e test execution time and the parsing is already covered by unit tests.

Reply:
I wasn't sure if the metric was broken for the maxint64 case based on the output:

> Sep 27 19:52:50.025: Expected
> : -9223372036854775808
> to equal
> : 256

Let me put it another way -- do you expect the metric to have a value that makes sense if someone configures a `/64` ?

Reply:
If someone configures a /64 or higher, the exposed value is MaxInt64 (as is being set here https://github.com/metallb/metallb/blob/997a177a762c6c8215ca9438083545529b675376/internal/allocator/allocator.go#L326)

I assumed the error was because of:
- the new ipv6 range, resulting in MaxInt64
- a bad int translation in the e2e infrastructure (translating to int instead of int64).

I ended up changing the pool range, because I thought e2e would be more meaningful with a regular case than into the corner case.

Now, I should have double tested it before assuming it was a bug in the tests. I'll validate it manually and fix it / cover it with a test if it's a bug.

Reply:
@fedepaol says he'll follow up with another fix if something is broken with the metric itself with a `/64`

Reply:
Just to add that I dug into this a bit more. The metric exposed to the endpoint looks fine:

`metallb_allocator_addresses_total{pool="dev-env-bgp"} 9.223372036854776e+18`

What's not working, is the fact that the parsing library returns it as a float64 and as part of our tests we are trying to convert it to int64, which doesn't really work as expected https://github.com/golang/go/issues/29463

Metrics wise, everything is fine and there's no bug I think.

Reply:
thank you for confirming!